Gemfile

```diff
@@ -2,39 +2,39 @@ source 'https://rubygems.org'
 ruby '2.7.6'
 
 # Essential gems: servers, adapters, Rails + Rails requirements
-gem 'coffee-rails', '~> 4.2.2'
-gem 'counter_culture', '~> 2.0'
-gem 'fastimage', '~> 2.1'
+gem 'coffee-rails', '~> 5.0.0'
+gem 'counter_culture', '~> 3.2' # Taeir: from v2 to v3 drops support for ruby 2.6
+gem 'fastimage', '~> 2.2'
 gem 'image_processing', '~> 1.12'
-gem 'jquery-rails', '~> 4.3.5'
-gem 'mysql2', '~> 0.5.3'
-gem 'puma', '~> 4.3.12'
-gem 'rails', '~> 5.2'
-gem 'rails-html-sanitizer', '~> 1.3'
-gem 'redis', '~> 4.1'
-gem 'rotp', '~> 6.0'
-gem 'sass-rails', '~> 5.0'
-gem 'tzinfo-data', '~> 1.2019.3'
-gem 'uglifier', '>= 1.3.0'
+gem 'jquery-rails', '~> 4.5.0'
+gem 'mysql2', '~> 0.5.4'
+gem 'puma', '~> 4.3.12' # Taeir: Check migration to puma 5
+gem 'rails', '~> 6.0.0'
+gem 'rails-html-sanitizer', '~> 1.4'
+gem 'redis', '~> 4.8'
+gem 'rotp', '~> 6.2'
+gem 'sass-rails', '~> 5.0' # Taeir: v6 is a wrapper around sassc-rails, which differs in behavior, verify to be working
+gem 'tzinfo-data', '~> 1.2022.3'
+gem 'uglifier', '>= 4.2.0'
 
 # Sign in
-gem 'devise', '~> 4.7'
-gem 'omniauth', '~> 1.9'
+gem 'devise', '~> 4.8'
+gem 'omniauth', '~> 1.9' # Taeir: Should be updated to 2.0+, requires testing?
 
 # Markdown support in both directions.
 gem 'commonmarker', '~> 0.23'
-gem 'reverse_markdown', '~> 2.0'
+gem 'reverse_markdown', '~> 2.1'
 
 # Charting stuff.
-gem 'chartkick', '~> 3.4'
-gem 'groupdate', '~> 4.3'
+gem 'chartkick', '~> 4.2' # Taeir: v3 -> v4 drops support for Ruby < 2.6
+gem 'groupdate', '~> 6.1' # Taeir: v4 -> v6 drops support for Ruby < 2.6
 
 # View stuff.
-gem 'diffy', '~> 3.3'
-gem 'jbuilder', '~> 2.10'
-gem 'rqrcode', '~> 1.1'
+gem 'diffy', '~> 3.4'
+gem 'jbuilder', '~> 2.11'
+gem 'rqrcode', '~> 2.1'
 gem 'will_paginate', '~> 3.3'
-gem 'will_paginate-bootstrap', '~> 1.0'
+gem 'will_paginate-bootstrap', '~> 1.0' # Taeir: unmaintained, last update 7 years ago...
```

Comment:
This is definitely unmaintained but AFAICT still works. However, we should review whether we're actually using this - I think our pagination may have been separated from this some time ago now that we use our own design framework but the dependency hasn't been removed? Not critical, can be left for now.

Reply:
I checked by removing it, and it is definitely still being used for the pagination of posts.